File: src/main/java/leetCode/t1342/java/Solution.java
package leetCode.t1342.java;

class Solution {
    public int numberOfSteps(int num) {
        int result = 0;
        while (num > 0) {
            if (num % 2 == 0) {
                num = num / 2;
            } else {
                num -= 1;
            }
            result++;
        }
        return result;
    }
}

File: src/main/java/leetCode/t1342/kotlin/Solution.kt
package leetCode.t1342.kotlin

class Solution {
    fun numberOfSteps(num: Int): Int {
        var res = 0
        var number = num
        while (number > 0) {
            if (number % 2 == 0) {
                number /= 2
            } else {
                number -= 1
            }
            res++
        }
        return res
    }
}